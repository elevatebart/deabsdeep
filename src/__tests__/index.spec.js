const deabsDeep = require('../index');

test('recursively replace absolute paths in keys and values', () => {
	const result = deabsDeep({
		[`${__dirname}/a/b.js`]: `${__dirname}/c/d.js`,
		foo: {
			[`${__dirname}/e/f.txt`]: `${__dirname}/g/h.txt`,
			bar: 1,
		},
	});
	expect(result).toMatchSnapshot();
});

test('recursively replace custom paths in keys and values', () => {
	const result = deabsDeep(
		{
			[`/foo/bar/a/b.js`]: `/foo/bar/c/d.js`,
			foo: {
				[`/foo/bar/e/f.txt`]: `/foo/bar/g/h.txt`,
				bar: {
					baz: 42,
				},
			},
		},
		{ root: '/foo/bar' }
	);
	expect(result).toMatchSnapshot();
});

test('absolute paths in array values', () => {
	const result = deabsDeep([
		`${__dirname}/a/b.js`,
		`${__dirname}/e/f.txt`,
		42,
		{
			bar: {
				baz: 42,
			},
		},
	]);
	expect(result).toMatchSnapshot();
});

test('custom mask', () => {
	const result = deabsDeep([`${__dirname}/a/b.js`], { mask: '<rootDir>' });
	expect(result).toMatchSnapshot();
});

test('ignore non plain objects', () => {
	class Pizza {
		constructor(x) {
			this.data = x;
		}
	}
	const result = deabsDeep({
		a: undefined,
		b: null,
		c: 1,
		d: 'x',
		e: new Date('1984-12-05T03:24:00'),
		f: /x/,
		g: new Pizza('x'),
	});
	expect(result).toMatchSnapshot();
});
