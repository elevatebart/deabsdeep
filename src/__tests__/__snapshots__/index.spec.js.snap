// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`absolute paths in array values 1`] = `
Array [
  "~/__tests__/a/b.js",
  "~/__tests__/e/f.txt",
  42,
  Object {
    "bar": Object {
      "baz": 42,
    },
  },
]
`;

exports[`custom mask 1`] = `
Array [
  "<rootDir>/__tests__/a/b.js",
]
`;

exports[`ignore non plain objects 1`] = `
Object {
  "a": undefined,
  "b": null,
  "c": 1,
  "d": "x",
  "f": /x/,
  "g": Pizza {
    "data": "x",
  },
}
`;

exports[`recursively replace absolute paths in keys and values 1`] = `
Object {
  "foo": Object {
    "bar": 1,
    "~/__tests__/e/f.txt": "~/__tests__/g/h.txt",
  },
  "~/__tests__/a/b.js": "~/__tests__/c/d.js",
}
`;

exports[`recursively replace custom paths in keys and values 1`] = `
Object {
  "foo": Object {
    "bar": Object {
      "baz": 42,
    },
    "~/e/f.txt": "~/g/h.txt",
  },
  "~/a/b.js": "~/c/d.js",
}
`;
